File: Examples/test-suite/php/cpp11_std_function_runme.php
<?php

require "tests.php";

$fn = pass_checker(420);

// invoke functor via call method
if (!$fn->call(420, "Petka"))
  check::fail("Petka 420 should be true");

if ($fn->call(419, "Chapai"))
  check::fail("Chapai 419 should be false");

// call wrapper
if (!call_function($fn, 420, "Petka"))
  check::fail("Petka 420 should be true");

if (call_function($fn, 419, "Chapai"))
  check::fail("Chapai 419 should be false");

// NULL handling
$passed = false;
try {
  call_function(null, 420, "Petka");
} catch (TypeError $e) {
  $passed = true;
}
if (!$passed)
  check::fail("call_function accepted invalid argument");

// Invalid argument
$passed = false;
try {
  call_function("invalid", 420, "Petka");
} catch (TypeError $e) {
  $passed = true;
}
if (!$passed)
  check::fail("call_function accepted invalid argument");

check::done();

File: Examples/test-suite/lua/cpp11_std_function_runme.lua
require("cpp11_std_function")

local fn = cpp11_std_function.pass_checker(420)

-- invoke functor via call method
assert(fn:call(420, "Petka"), "Petka 420 should be true")
assert(not fn:call(419, "Chapai"), "Chapai 419 should be false")

-- call wrapper
assert(cpp11_std_function.call_function(fn, 420, "Petka"), "Petka 420 should be true")
assert(not cpp11_std_function.call_function(fn, 419, "Chapai"), "Chapai 419 should be false")

-- NULL handling
assert(not pcall(function()
  cpp11_std_function.call_function(nil, 420, "Petka")
end), "call_function accepted invalid argument")

-- Invalid argument
assert(not pcall(function()
  cpp11_std_function.call_function("invalid", 420, "Petka")
end), "call_function accepted invalid argument")
